### ISSUE TITLE
ESBMC crash - ERROR: Boolector error encountered

### ISSUE THREAD

**Issue by tihanyin** [esbmc/esbmc#1067]
In this small C example ESBMC 7.2 crashes with `ERROR: Boolector error encountered [boolector] boolector_write: element bit-width of 'e_array' and bit-width of 'e_value' must be equal`

```c
//FormAI DATASET v1.0. Category: Memory management ; Style: calm
#include<stdio.h>
#include<stdlib.h>

int main(){
    int* ptr;
    int size;
    printf("Enter number of integers: ");
    scanf("%d", &size);

    // Allocating memory dynamically for array
    ptr = (int*) malloc(size * sizeof(int));
    if(ptr == NULL){
        printf("Memory allocation failed.\n");
        exit(0);
    }

    // Initializing the array
    for(int i = 0; i < size; i++){
        *(ptr + i) = i + 1;
    }

    // Displaying the array
    printf("The array is: ");
    for(int i = 0; i < size; i++){
        printf("%d ", *(ptr + i));
    }

    // Reallocating memory for array
    printf("\nEnter new size of array: ");
    scanf("%d", &size);
    ptr = realloc(ptr, size * sizeof(int));
    if(ptr == NULL){
        printf("Memory reallocation failed.\n");
        exit(0);
    }

    // Adding new elements to the array
    printf("Enter %d new elements of the array:\n", size);
    for(int i = size - 5; i < size; i++){
        scanf("%d", &*(ptr + i));
    }

    // Displaying the new array
    printf("The new array is: ");
    for(int i = 0; i < size; i++){
        printf("%d ", *(ptr + i));
    }

    // Freeing the allocated memory
    free(ptr);
    printf("\nMemory freed successfully.\n");
    return 0;
}
``` 

ESBMC crash output:

``` 
esbmc --unwind 1 DATA3/FormAI_31658.c
ESBMC version 7.2.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing
Converting
Generating GOTO Program
GOTO program creation time: 0.260s
GOTO program processing time: 0.004s
Starting Bounded Model Checking
Not unwinding 
Not unwinding 
Not unwinding 
Not unwinding 
Symex completed in: 0.162s (90 assignments)
Slicing time: 0.001s (removed 40 assignments)
Generated 20 VCC(s), 18 remaining after simplification (50 assignments)
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector/floating-point arithmetic
ERROR: Boolector error encountered
[boolector] boolector_write: element bit-width of 'e_array' and bit-width of 'e_value' must be equal

Aborted (core dumped)
```

**Comment by lucasccordeiro** [comment]
@fbrausse: Here is another issue related to the SMT backend. Can I ask you please to check it?

**Comment by fbrausse** [comment]
Fixed by 69841f931f3865f6384bb07174be1479987ea057, which is part of #905.

**Comment by lucasccordeiro** [comment]
Thanks, @fbrausse. 
?
BTW, shall we get https://github.com/esbmc/esbmc/pull/905 merged soon? Is there something else you plan to implement
